tests/test_ComputeFeatures.py

```diff
@@ -493,6 +494,35 @@ def test_ComputeFeatures_generate_RSCU():
     }
 
 
+def test_ComputeFeatures_compute_feature():
+    """Test all pair properties are populated correctly from running compute_feature()."""
+    test_CF = ComputeFeatures(
+        test_virus_genome_dir,
+        test_host_genome_dir,
+        test_virus_gene_dir,
+        test_host_gene_dir,
+    )
+    test_CF.add_blastn_files(
+        "tests/datatests/blastn_phagevhost.tsv",
+        "tests/datatests/blastn_phagevspacer.tsv",
+    )
+    test_CF.do_setup()  # determining all possible virus-host pairs, gets fasta headers, read and process blastn_output, computes GC content and k-mer profiles, and for each organism: generate dictionaries of codon, amino acid, and synonymous codon usage frequencies.
+    test_CF.compute_feature(
+        test_CF.pairs[0]
+    )  # computes comparisons (e.g. distances) between profiles generated in do_setup for a single virus and single host
+
+    # Genome-level features
+    assert isinstance(test_CF.pairs[0].GCdifference, float)
+    assert isinstance(test_CF.pairs[0].k3dist, float)
+    assert isinstance(test_CF.pairs[0].k6dist, float)
+    assert isinstance(test_CF.pairs[0].homology_hit, bool)
+
+    # Gene-level features
+    assert isinstance(test_CF.pairs[0].codons_comparison, CodonBiasComparison)
+    assert isinstance(test_CF.pairs[0].aa_comparison, CodonBiasComparison)
+    assert isinstance(test_CF.pairs[0].RSCU_comparison, CodonBiasComparison)
+
```

Comment:
can you add tests where you also check you get the right values? I can imagine a change in the algorithm to compute R2 would change the values but this test would not capture it since it only check for instances.

Reply:
Still working on test implementation without errors. But per discussion, I implemented tests on the values for extra overlap. I also wrote another separate test after implementing pairs_of_interest as a ComputeFeatures class attribute for use in determine_custom_pairs() - having this method made it easier to compute and test values for a single pair without the use of a next() function or a for loop through the all the pairs in our pre-existing virus and host sequence directories.